File: python_leaoai/python_leaoai/repositories/Implementations/Factory/EmbalagemTypeFactory.py
# repositories/embalagem_type_factory.py
from sqlalchemy.orm import Session
from typing import List, Optional
from models.factory import EmbalagemType

class EmbalagemTypeFactory:
    def __init__(self, db: Session):
        self.db = db

    # --- CRUD ---
    def get_obj(self, id: int) -> Optional[EmbalagemType]:
        return self.db.query(EmbalagemType).filter(EmbalagemType.id == id).first()

    def save_obj(self, obj: EmbalagemType) -> EmbalagemType:
        self.db.add(obj)
        self.db.commit()
        self.db.refresh(obj)
        return obj

    def update_obj(self, obj: EmbalagemType) -> EmbalagemType:
        self.db.merge(obj)
        self.db.commit()
        self.db.refresh(obj)
        return obj

    def get_all(self) -> List[EmbalagemType]:
        return self.db.query(EmbalagemType).all()

    def get_list_actives(self) -> List[EmbalagemType]:
        return self.db.query(EmbalagemType).filter(EmbalagemType.ativo == True).all()

    def get_list(self) -> List[EmbalagemType]:
        return self.db.query(EmbalagemType).filter(EmbalagemType.excluido == False).all()

    def delete_obj(self, id: int) -> bool:
        obj = self.get_obj(id)
        if not obj:
            return False
        try:
            obj.ativo = False
            obj.excluido = True
            self.db.commit()
            return True
        except:
            self.db.rollback()
            return False

File: Framework.Repositories/Implementations/Factory/EmbalagemTypeFactory.cs
using Data;
using Framework.Data.Models.Factory;
using System.Collections.Generic;
using System.Linq;

namespace Framework.Repositories.Implementations.Factory
{
    public class EmbalagemTypeFactory
    {
        private readonly ApplicationDbContext _context;

        public EmbalagemTypeFactory(ApplicationDbContext context)
        {
            _context = context;
        }

        #region Crud
        public EmbalagemType GetObj(int id)
        {
            return _context.EmbalagemTypes.FirstOrDefault(x => x.Id == id);
        }

        public EmbalagemType SaveObj(EmbalagemType obj)
        {
            _context.EmbalagemTypes.Add(obj);
            _context.SaveChanges();
            return obj;
        }

        public EmbalagemType UpdateObj(EmbalagemType obj)
        {
            _context.EmbalagemTypes.Update(obj);
            _context.SaveChanges();
            return obj;
        }

        public List<EmbalagemType> GetAll()
        {
            return _context.EmbalagemTypes.ToList();
        }
        public List<EmbalagemType> GetListActives()
        {
            return _context.EmbalagemTypes.Where(x => x.Ativo).ToList();
        }
        public List<EmbalagemType> GetList()
        {
            return _context.EmbalagemTypes.Where(x => !x.Excluido).ToList();
        }

        public bool DeleteObj(int id)
        {
            var obj = GetObj(id);
            if (obj == null)
                return false;
            try
            {
                obj.Ativo = false;
                obj.Excluido = true;
                _context.EmbalagemTypes.Update(obj);
                _context.SaveChanges();
                return true;
            }
            catch (Exception)
            {
                return false;
            }
        }
        #endregion
    }
}
